service/src/main/java/com/codeforcommunity/processor/authenticated/ProtectedSchoolProcessorImpl.java

```diff
@@ -96,10 +98,58 @@ public School createSchool(JWTData userdata, NewSchoolRequest newSchoolRequest)
           school.getHidden(),
           contacts);
     }
-
     throw new SchoolAlreadyExistsException(name, country);
   }
 
+  @Override
+  public void updateSchool(
+      JWTData userData, int schoolId, UpsertSchoolRequest upsertSchoolRequest) {
+    SchoolsRecord school = db.selectFrom(SCHOOLS).where(SCHOOLS.ID.eq(schoolId)).fetchOne();
```

Comment:
Lets update this query to check `.and(SCHOOL.DELETED_AT.isNull())` to make sure we don't consider a "soft deleted" school here